### PR TITLE
refactor: factor out CDP message types

### PR DESCRIPTION
**Pull request by crowlKats**

### CODE REVIEW

File: cli/inspector_structures.rs [hunk 1]
```diff
@@ -0,0 +1,425 @@
+use deno_core::serde_json;
```
Comment (bartlomieju) [inline cli/inspector_structures.rs:1]: Missing copyright; consider renaming to `cdp.rs` and adding link to https://chromedevtools.github.io/devtools-protocol/tot/

File: cli/tools/coverage/mod.rs [hunk 2]
```diff
@@ -43,33 +43,32 @@ impl CoverageCollector {
   }
 
   async fn enable_debugger(&mut self) -> Result<(), AnyError> {
-    self.session.post_message("Debugger.enable", None).await?;
+    self.session.post_message("Debugger.enable", ()).await?;
```
Comment (bartlomieju) [inline cli/tools/coverage/mod.rs:46]: Why is this change necessary? `None` reflects intent better when there are no parameters than `()`; could we revert back to the previous API?

File: cli/tools/repl/channel.rs [hunk 3]
```diff
@@ -34,14 +35,8 @@ pub fn rustyline_channel(
...
-  PostMessage {
-    method: String,
-    params: Option<Value>,
-  },
-  LspCompletions {
-    line_text: String,
-    position: usize,
-  },
+  PostMessage { method: String, params: Value },
+  LspCompletions { line_text: String, position: usize },
```
Comment (bartlomieju) [inline cli/tools/repl/channel.rs:39]: Seems like unrelated change

File: cli/inspector_structures.rs [hunk 4]
```diff
@@ -0,0 +1,425 @@
...
+use deno_core::serde_json::Value;
+use serde::Deserialize;
+use serde::Serialize;
+
+#[derive(Debug, Clone, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct AwaitPromiseArgs {
```
Comment (bartlomieju) [inline cli/inspector_structures.rs:8]: Could you add links to relevant message in https://chromedevtools.github.io/devtools-protocol/tot/ as docstrings?

File: cli/tools/repl/channel.rs (hunk 3 above)
Comment (crowlKats) [reply cli/tools/repl/channel.rs:39]: nope, since i the `Option<>` from `params` it reformatted it all

**Review by lucacasonato — APPROVED** [review]
LGTM

**Comment by bartlomieju** [comment]
Maybe rename to `refactor: factor out CDP message types`?

**Comment by ry** [comment]
What does this feature do?

**Comment by bartlomieju** [comment]
> What does this feature do?

Not a feature, refactor that is a prerequisite for https://github.com/denoland/deno/pull/12899
